sycl/test-e2e/Regression/no-split-reqd-wg-size.cpp

```diff
@@ -0,0 +1,39 @@
+// This test checks that with -fsycl-device-code-split=off, two kernels
+// with different reqd_work_group_size dimensions can be launched.
+// RUN: %{build} -fsycl -fsycl-device-code-split=off -o %t.out
+// RUN: %{run} %t.out
+#include <sycl/sycl.hpp>
+
+constexpr int WGSIZE = 4;
+
+using namespace sycl;
+
+void kernel_launch_2(queue &q) {
+  range<1> globalRange(WGSIZE);
+  range<1> localRange(WGSIZE);
+  nd_range<1> NDRange(globalRange, localRange);
+  q.submit([&](handler &cgh) {
+     cgh.parallel_for<class testNDRange2>(
+         NDRange, [=](nd_item<1> it) [[sycl::reqd_work_group_size(WGSIZE)]] {});
+   }).wait();
+}
+
+void kernel_launch(queue &q) {
+  range<2> globalRange(WGSIZE, WGSIZE);
+  range<2> localRange(WGSIZE, WGSIZE);
+  nd_range<2> NDRange(globalRange, localRange);
+  q.submit([&](handler &cgh) {
+     cgh.parallel_for<class testNDRange>(
+         NDRange,
+         [=](nd_item<2> it) [[sycl::reqd_work_group_size(WGSIZE, WGSIZE)]] {});
```

Comment:
Not sure exactly, but can `sycl::reqd_work_group_size(WGSIZE, WGSIZE)` and `sycl::reqd_work_group_size(WGSIZE)` be moved to some template or function parameter (+ move dimension to the template parameter), to not duplicate the same code in both `kernel_launch_*` functions.

Reply:
Unfortunately, it seems attributes cannot accept parameter packs, so the best we can do is the macro version:

```c++
#define TEST(...)                                                              \
  {                                                                            \
    range globalRange(__VA_ARGS__);                                            \
    range localRange(__VA_ARGS__);                                             \
    nd_range NDRange(globalRange, localRange);                                 \
    q.parallel_for(NDRange,                                                    \
                   [=](auto) [[sycl::reqd_work_group_size(__VA_ARGS__)]] {});  \
  }
```

Reply:
> Unfortunately, it seems attributes cannot accept parameter packs

We should create a tracker for that, because it is possible to add such support and we have done it already for other attributes

Reply:
Created an issue: https://github.com/intel/llvm/issues/13686